src/project.py

```diff
@@ -217,7 +229,118 @@ def get_tps_time(outfiles):
+        else:
+            atom_positions = atom_type_pos(frame, atom_type)
+        positions.append(atom_positions)
+    msd = freud.msd.MSD(box=trajectory[-1].configuration.box, mode=msd_mode)
+    msd.compute(positions)
+    f.close()
+    return(msd.msd)
 
+@directives(ngpu=1)
+@on_pflow
+@MyProject.operation
+@MyProject.post(rdfed)
+@MyProject.pre(sampled)
+def post_proc(job):
+    import cmeutils
+    from cmeutils.structure import gsd_rdf, get_quaternions
+    import numpy as np
+    import os
+    import matplotlib
+    import matplotlib.pyplot as plt
+    import gsd
+    import gsd.hoomd
+    import freud
+
+    def msd_from_gsd(gsdfile, start=-30, stop=-1, atom_type='c', msd_mode = "window"):
+        from gsd import pygsd
+        import freud
+        f = gsd.pygsd.GSDFile(open(gsdfile, "rb"))
+        trajectory = gsd.hoomd.HOOMDTrajectory(f)
+        positions = []
+        for frame in trajectory[start:stop]:
+        	if atom_type == 'all':
+            		atom_positions = frame.particles.position[:]
+        	else:
+            		atom_positions = atom_type_pos(frame, atom_type)
+        	positions.append(atom_positions)
+        msd = freud.msd.MSD(box=trajectory[-1].configuration.box, mode=msd_mode)
+        msd.compute(positions)
+        f.close()
+        return(msd.msd)
+    def atom_type_pos(snap, atom_type):
+    	if not isinstance(atom_type, list):
+        	atom_type = [atom_type]
+    	positions = []
+    	for atom in atom_type:
+        	indices = np.where(snap.particles.typeid == snap.particles.types.index(atom))
+        	positions.append(snap.particles.position[indices])
+    	return np.concatenate(positions)
+
+    gsdfile= job.fn('trajectory.gsd')
+    with gsd.hoomd.open(gsdfile) as f:
+        snap= f[0]
+        all_atoms=snap.particles.types
+        os.makedirs(os.path.join(job.ws,"rdf/rdf_txt"))
+        os.makedirs(os.path.join(job.ws,"rdf/rdf_png"))
+        os.makedirs(os.path.join(job.ws,"msd/msd_array"))
+        os.makedirs(os.path.join(job.ws,"msd/msd_png"))
+        os.makedirs(os.path.join(job.ws,"diffraction/diffraction_plots"))
+        for types in all_atoms:
+            A_name=types
+            B_name=types
+            rdf,norm = gsd_rdf(gsdfile,A_name, B_name, r_min=0.01, r_max=5)
+            x = rdf.bin_centers
+            y = rdf.rdf*norm
+            save_path= os.path.join(job.ws,"rdf/rdf_txt/{}_rdf.txt".format(A_name))
+            np.savetxt(save_path, np.transpose([x,y]), delimiter=',', header= "bin_centers, rdf")
+            plt.xlabel("r (A.U.)", fontsize=14)
+            plt.ylabel("g(r)", fontsize=14)
+            plt.plot(x, y)
+            save_plot= os.path.join(job.ws,"rdf/rdf_png/{}_rdf.png".format(A_name))
+            plt.savefig(save_plot)
+            msd_array= msd_from_gsd(gsdfile, start=-30, stop=-1, atom_type=A_name, msd_mode = "window")
+            save_path= os.path.join(job.ws, "msd/msd_array/{}.npy".format(A_name))
```

Comment:
I would definitely prefer .txt files but I wasn't sure if it was possible to convert a .npy file to a .txt file. Also, this msd_from_gsd code we are using creates a numpy array; Is there a way to have it make a text file instead?

Reply:
Well, the numpy array that's retruned by `msd_from_gsd` is just stored as a python object, and we can save it to a text file using np.savetxt just like you have already been using. `gsd_rdf` also returns a numpy array.

The way you enforce it's saved as a `.txt` rather than a `.npy` is by just making the file name a `txt`

Reply:
Saving the results of `msd_from_gsd` could follow the exact same steps as saving the RDF

```
save_path= os.path.join(job.ws,"rdf/rdf_txt/{}_rdf.txt".format(A_name))
np.savetxt(save_path, np.transpose([x,y]), delimiter=',', header= "bin_centers, rdf")
```

Maybe it would look something like this:
```
save_path= os.path.join(job.ws,"msd/msd_txt/{}_msd.txt".format(A_name))
np.savetxt(save_path, msd_array, delimiter=',', header= "msd")
```

Reply:
ok did this, thanks this is awesome!